processing/src/main/java/org/apache/druid/query/aggregation/first/StringFirstAggregatorFactory.java

```diff
@@ -188,7 +188,7 @@ public Comparator getComparator()
   @Override
   public Object combine(Object lhs, Object rhs)
   {
-    return TIME_COMPARATOR.compare(lhs, rhs) > 0 ? lhs : rhs;
+    return TIME_COMPARATOR.compare(lhs, rhs) < 0 ? lhs : rhs;
```

Comment:
wow! Is there a test for this somewhere?

Reply:
Yes. The existing test case was testing the wrong behavior. The fix to the test is also in this PR.